lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1369,6 +1369,10 @@ abstract class DefaultTextEditingStrategy with CompositionAwareMixin implements
       final DomKeyboardEvent event = e as DomKeyboardEvent;
       if (event.keyCode == _kReturnKeyCode) {
         onAction!(inputConfiguration.inputAction);
+        // Stop key event propagation if the input type is not multiline.
```

Comment:
`e.stopPropagation()` is different from `e.preventDefault()`, and your comment could be confused for the former.

You could phrase as follows: "Prevent the browser from inserting a new line when it's not a multiline input."